cdap-archetypes/cdap-spark-java-archetype/src/main/resources/archetype-resources/src/main/java/SparkPageRankApp.java

```diff
@@ -93,15 +127,27 @@ public SparkSpecification configure() {
    */
   public static final class RanksServiceHandler extends AbstractHttpServiceHandler {
 
+    private static final Gson GSON = new Gson();
+    public static final String URL_KEY = "url";
+    public static final String RANKS_SERVICE_PATH = "rank";
+
     @UseDataSet("ranks")
     private ObjectStore<Integer> ranks;
 
-    @Path("rank")
-    @GET
-    public void getRank(HttpServiceRequest request, HttpServiceResponder responder, @QueryParam("url") String url) {
+    @Path(RANKS_SERVICE_PATH)
+    @POST
```

Comment:
We need to use POST call as the query string being sent has / and : which can't be sent through path params and the UI does not support query parameters. 
I renamed the method and removed get from it for some clarity.